pkg/apis/serving/k8s_validation.go

```diff
@@ -320,7 +322,7 @@ func validateSidecarContainer(ctx context.Context, container corev1.Container, v
 		errs = errs.Also(apis.CheckDisallowedFields(*container.ReadinessProbe,
 			*ProbeMask(&corev1.Probe{})).ViaField("readinessProbe"))
 	}
-	return errs.Also(validate(ctx, container, volumes))
+	return errs.Also(validate(ctx, container, volumes, true /*isSidecar*/))
```

Comment:
The comments are useful, but in other contexts (pun intended) we decorate context to pass down this sort of information to avoid polluting the signature.  For example:
```
	return errs.Also(validate(serving.WithinUserContainer(ctx), container, volumes))
```

vs.

```
	return errs.Also(validate(serving.WithinSidecarContainer(ctx), container, volumes))
```

Then on the consumer side, you can check:

```
   if serving.IsInSidecar(ctx) { ... }
```

I can point you to a few places we do this, but this is nice because it percolates down to everywhere we've plumbed context without needing to plumb `isSidecar` all of those places.

Reply:
Ok I can do that.

Reply:
@mattmoor looking at this again I think that there are trade offs as if this is an optional parameter it should be part of the signature as bugs can be introduced if someone forgets to pass the context.